packages/autorest.go/src/generator/generator.ts

```diff
@@ -108,10 +108,17 @@ export async function generateCode(host: AutorestExtensionHost) {
     }
 
     const responses = await generateResponses(session.model);
-    if (responses.length > 0) {
+    if (responses.responses.length > 0) {
       host.writeFile({
         filename: `${filePrefix}responses.go`,
-        content: responses,
+        content: responses.responses,
+        artifactType: 'source-file-go'
+      });
+    }
+    if (responses.serDe.length > 0) {
+      host.writeFile({
+        filename: `${filePrefix}responses_serde.go`,
```

Comment:
Technically, there's no "ser" for response envelopes, but `responses_de.go` seems weird.

Reply:
Right, at first glance I'd guess that's German responses